shared/src/main/java/dice/RollResult.java

```diff
@@ -1,39 +1,114 @@
-package dice;
-
-import java.util.ArrayList;
-
-public class RollResult {
-    private ArrayList<Integer> rolls;
-
-    public RollResult() {
-        rolls = new ArrayList<Integer>();
-    }
-
-    public void addRoll(int rollValue) {
-        rolls.add(rollValue);
-    }
-
-    public int sum() {
-        int sum = 0;
-
-        for (int rollIndex = 0; rollIndex < rolls.size(); rollIndex++) {
-            sum += rolls.get(rollIndex);
-        }
-
-        return sum;
-    }
-
-    public int rollCount() {
-        return rolls.size();
-    }
-
-    public int[] rollsArray() {
-        int[] array = new int[rolls.size()];
-        int index = 0;
-        for (Integer n : rolls) {
-            array[index] = n;
-            index++;
-        }
-        return array;
-    }
-}
+package dice;
+
+import java.util.ArrayList;
+
+public class RollResult {
+    private ArrayList<Integer> rolls;
+    private String specialRoll;
+    private int specialRollCode;
+
+    public RollResult() {
+        rolls = new ArrayList<Integer>();
+        specialRoll = "";
+        specialRollCode=0;
+    }
+
+    public void addRoll(int rollValue) {
+        rolls.add(rollValue);
+        updateSpecialRoll();
+    }
+
+    public int sum() {
+        int sum = 0;
+
+        for (int rollIndex = 0; rollIndex < rolls.size(); rollIndex++) {
+            sum += rolls.get(rollIndex);
+        }
+
+        return sum;
+    }
+
+    public int rollCount() {
+        return rolls.size();
+    }
+
+    public int[] rollsArray() {
+        int[] array = new int[rolls.size()];
+        int index = 0;
+        for (Integer n : rolls) {
+            array[index] = n;
+            index++;
+        }
+        return array;
+    }
+    
+    public String getSpecialRollString(){
+        return specialRoll;
+    }
+    
+    public int getSpecialRollCode(){
+        return specialRollCode;
+    }
+    
+    private void updateSpecialRoll(){
+        ArrayList<Integer> temp = new ArrayList(rolls);
+        if(temp.size() == 3){

```

Comment:
Can you make your bracket usage consistent in this method. Preferably make it match existing code in the file